feat[ft-search-product]

ngoma3 opened the pull request:
### What does this PR do

-  allow user to search product
-   write its swagger documentation
### How to run it

- git clone from repo
- git checkout ft-search-product
- set .env as shown in .envexample
-  run npm i
-  run npm run dev
-  copy this url in your postman: http://localhost:{port}/api/v1/product
-  you will add query keywords on this GET url `ex`: http://localhost:{port}/api/v1/product?q=apple
-  the api will search for the alike content in the product collection depending on name,description and model
-  it will respond the arrayList with all matched products.

Kevin-ishimwe commented on src/controllers/products/productController.js:
```diff
@@ -97,3 +99,29 @@ export const deleteProduct = asyncWrapper(async (req, res) => {
+export const search = async (req, res) => {
+  console.log("this is where the error should");
+  const { q } = req.query;
+  console.log("this is where the error should occur",q);
```
and this one to

Kevin-ishimwe commented on src/controllers/products/productController.js:
```diff
@@ -97,3 +99,29 @@ export const deleteProduct = asyncWrapper(async (req, res) => {
   });
 });
+export const search = async (req, res) => {
+  console.log("this is where the error should");
```
if possible could we remove these console.log()

FabCode67 commented on src/controllers/products/productController.js:
```diff
@@ -97,3 +99,29 @@ export const deleteProduct = asyncWrapper(async (req, res) => {
+    },
+  });
+
+  return res.send({status: req.t('success'), itemsList: items});
```
add status code here?

FabCode67 commented on src/controllers/products/productController.js:
```diff
@@ -97,3 +99,29 @@ export const deleteProduct = asyncWrapper(async (req, res) => {
+  });
+
+  return res.send({status: req.t('success'), itemsList: items});
+} catch (err) {
```
remove this catch block  by importing asyncWrapper middleware

Evaristekanova reviewed (dismissed):
@ngoma3 try to remove try and catch and use the reusable function to handle that and work on the way you are sending responses cz you are not sending status try to use jsend response

Christabll reviewed (commented):
I think You should  have proper status message

BenoitIT commented:
There is a conflict on your PR in reademe

Commit:
feat[ft-search-product]
This branch will allow a user to search product by using different keywords typed in the searchbar on the front end

- allow user to search product
- write its swagger documentation

- git clone from repo
- git checout ft-password-rese
- set .env as shown in .envexample
- run npm i
- run npm run dev
- copy this url in your postman: http://localhost:{port}/api/v1/product
- you will add query keywords on this GET url ex: http://localhost:{port}/api/v1/product?q=apple
- the api will search for the alike content in the product collection depending on name,description and model
- it will respond the arrayList with all matched products.

feat[WIP ft-search-product]

feat[ft-search-product]
This branch will allow a user to search product by using different keywords typed in the searchbar on the front end

- allow user to search product
- write its swagger documentation

- git clone from repo
- git checout ft-password-rese
- set .env as shown in .envexample
- run npm i
- run npm run dev
- copy this url in your postman: http://localhost:{port}/api/v1/product
- you will add query keywords on this GET url ex: http://localhost:{port}/api/v1/product?q=apple
- the api will search for the alike content in the product collection depending on name,description and model
- it will respond the arrayList with all matched products.

feat[WIP ft-search-product]

feat[ft-search-product]
This branch will allow a user to search product by using different keywords typed in the searchbar on the front end

- allow user to search product
- write its swagger documentation

- git clone from repo
- git checout ft-password-rese
- set .env as shown in .envexample
- run npm i
- run npm run dev
- copy this url in your postman: http://localhost:{port}/api/v1/product
- you will add query keywords on this GET url ex: http://localhost:{port}/api/v1/product?q=apple
- the api will search for the alike content in the product collection depending on name,description and model
- it will respond the arrayList with all matched products.

feat[WIP ft-search-product]

FabCode67 reviewed (approved):
Good job @ngoma3

Aimeana100 reviewed (approved):
Good job @ngoma3

mansurissa reviewed (approved):
LGTM
